app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1882,6 +1883,49 @@ class PluginApi {
   registerNotificationTypeRenderer(notificationType, func) {
     registerNotificationTypeRenderer(notificationType, func);
   }
+
+  /**
+   * EXPERIMENTAL. Do not use.
+   * Registers a new tab in the user menu. This API method expects a callback
+   * that should return a class inheriting from the class (UserMenuTab) that's
+   * passed to the callback. See discourse/app/lib/user-menu/tab.js for
+   * documentation of UserMenuTab.
```

Comment:
I had a look at `UserMenuTab` and we don't really have docs in there yet. Might be good to add it with this commit.

Reply:
Oops sorry I didn't see the chnages below when making this comment.